src/components/complex/Proposal/index.jsx

```diff
@@ -239,6 +243,87 @@ const Proposal = ({
     }
   }, [noData, activeChainId])
 
+  const handleMessageFromIframe = (event) => {
+    // Check the origin of the message for security
+    if (event.origin !== window.location.origin) {
+      console.error('Received an untrusted message from:', event.origin)
+      return
+    }
+
+    // Log or handle the received message
+    if (event.data.type === 'linkClick') {
+      if (event.data.id !== id) return
+      setReadMoreContent(event.data.href)
+      setLocation({ current: event.data.href, old: location.current })
+    }
+  }
+
+  const goBack = () => {
+    setReadMoreContent(location.old)
+    setLocation({ current: location.old, old: url })
+  }
+
+  useEffect(() => {
+    window.addEventListener('message', handleMessageFromIframe)
+
+    return () => {
```

Comment:
The return value here is a function. Where is it used?

Reply:
It is called when the proposal element is closed removing the listener. For example when loading a different page than overview the Proposal element is removed and this function called. Check https://react.dev/learn/lifecycle-of-reactive-effects#the-lifecycle-of-an-effect